sqlglot/dialects/presto.py

```diff
@@ -256,6 +256,19 @@ class Presto(Dialect):
     # https://github.com/prestodb/presto/issues/2863
     NORMALIZATION_STRATEGY = NormalizationStrategy.CASE_INSENSITIVE
 
+    # The result of certain math functions in Presto/Trino is of type
+    # equal to the input type e.g: FLOOR(5.5/2) -> DECIMAL, FLOOR(5/2) -> BIGINT
+    ANNOTATORS = {
+        **Dialect.ANNOTATORS,
+        exp.Floor: lambda self, e: self._annotate_by_args(e, "this"),
+        exp.Ceil: lambda self, e: self._annotate_by_args(e, "this"),
+        exp.Mod: lambda self, e: self._annotate_by_args(e, "this"),
+        exp.Round: lambda self, e: self._annotate_by_args(e, "this"),
+        exp.Sign: lambda self, e: self._annotate_by_args(e, "this"),
+        exp.Abs: lambda self, e: self._annotate_by_args(e, "this"),
+        exp.Rand: lambda self, e: self._annotate_by_args(e, "this"),
```

Comment:
Not sure about this one. Are we generating `RANDOM` or `RAND` for `exp.Rand`? See [docs](https://prestodb.io/docs/current/functions/math.html#rand):

<img width="259" alt="Screenshot 2024-07-21 at 5 51 25 PM" src="https://github.com/user-attachments/assets/c6c81982-841c-48ee-a7e0-efc6ad086371">

Reply:
We're generating both as `RAND()` currently; Will add `Type.DOUBLE` annotation as a fallback to the no arg case